### PR TITLE
Feature views

### PR DESCRIPTION
**Pull request by edahigure**
### Project requirements
- Implement the design from the sneak peek wireframes.
    - Styling is nice to have but focus on making sure everything functions first. (Note: Ignore login logic, you would be using Devise in coming milestone)

**Note: Analyze the wireframes deeply and try to identify all the necessary elements and how a user would navigate from one page to another**
    

blog_root_users (1)

blog_user_page

blog_user_all_posts

blog_single_post

- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).
- **Hint:** In order to see some posts or comments, you might need to add some data to your database as described [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md#use-models-to-insert-data).

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,2 +1,34 @@
...
+        <div class= "container-column-2"> 
+            <div class="item">Post # <%= post.id %>  <%= post.title %> </div>
+            <div class="item"> <%= post.text %> </div>
+
+            <div class="container-2">               
+              <div>Comments: <%= post.comments_counter %> </div>
+              <div>likes: <%= post.likes_counter %> </div>
+            </div>
+        </div>
+    <% end %>
```
Comment (Qoosim) [inline app/views/users/show.html.erb]: - [ ] According to the project requirements, You're expected to display the most recent posts here in your user's show page. But the arrangement of posts based on how recent they are is not consistent. I created four posts for the the second user created. If I navigate to the user's show page, the arrangement of the posts should be from the last one which is post four to the first post. [i.e 4 ... 1]. That's not the arrangement in your own case. Kindly check below video to confirm what I am saying.

[recent-post.webm](https://github.com/edahigure/blog-app-2/assets/26597432/f18c5d90-be32-4083-8ec2-968dae711c92)
 
Your can check below snippet for some insight :point_down: 

```
  <% @user.recent_post.each do |post| %>

      <h4><%= link_to post.title, user_post_path(post) %></h4>
      <p><%= post.text %></p>
      <p>
         <small>Comments: <%= post.comments_counter || 0 %>, 
          </small> 
          <small>Likes: <%= post.likes_counter || 0 %></small>
      </p>

<% end %>

```

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -1 +1,13 @@
...
+<div class="wrapper-1">
+  
+  <% @users.each do |user| %>
+    <li class="item">  
+       <img class="photo" src="<%= user.photo%>" alt="user photo">
+       <div class= "container"> 
+          <div class="name"> <%= user.name %> </div>
+          <div class="post-num"> number of posts: <%= user.posts_counter %></div>
+       </div>
+    </li>
```
Comment (Qoosim) [inline app/views/users/index.html.erb]: - [ ] It is required according to the project requirement that when a user name is clicked, it should redirect to the user's show page. This is not so in your app. Kindly look into it and fix.

[user-link.webm](https://github.com/edahigure/blog-app-2/assets/26597432/0ddca28d-f839-45c3-8a03-53d4cad7b82a)

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1,2 +1,34 @@
...
+
+
+    
+
+    <button type="button" class="button-1">See all posts</button>
+
```
Comment (Qoosim) [inline app/views/users/show.html.erb]: - [ ] It is required that `See all posts` link opens user's post index as displayed on this below wireframe.

### Wireframe
wireframe

### App Behaviour
[see-all-post.webm](https://github.com/edahigure/blog-app-2/assets/26597432/e0119c9a-9104-4ead-9cb9-e34d2edd7f69)

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1 +1,33 @@
...
+  
+      <div class= "container-column-2"> 
+        <% com =  @comments.where(post_id: post.id) %>
+        <% if com!= nil  %>
+            <% com.each do |comment|  %>
+            <div> <%= @users.find(comment.author_id).name %>:  <%= comment.text %> </div>
+            <% end %>
+        <% end %>        
+      </div>
+    <% end %>
```
Comment (Qoosim) [inline app/views/posts/index.html.erb]: - [ ] It is required that a post should render 5 most recent comments. This is not so in your case. Kindly look into it and fix.

You can check the code below for some insights.

```
   <% post.recent_comments.each do |comment| %>
       <p><%= comment.author.name %>: <%= comment.text %></p> 
    <% end %>
```

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @edahigure, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Implemented some project requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
## Hi @edahigure 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes

## To highlight :champagne: 
- Nice use of routes  :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**